base/range.jl

```diff
@@ -1388,14 +1388,12 @@ function sum(r::AbstractRange{<:Real})
 end
 
 function _in_range(x, r::AbstractRange)
-    if !isfinite(x)
-        return false
-    elseif iszero(step(r))
-        return !isempty(r) && first(r) == x
-    else
-        n = round(Integer, (x - first(r)) / step(r)) + 1
-        return n >= 1 && n <= length(r) && r[n] == x
-    end
+    isempty(r) && return false
+    f, l = first(r), last(r)
+    # check for NaN, Inf, and large x that may overflow in the next calculation
+    f <= x <= l || l <= x <= f || return false
```

Comment:
The current change will break `0. in (0. .* (1:10))`
We need to keep the `iszero(step(r))` branch or add `f == x && return true`

Reply:
Thanks! I'm starting to understand what "ranges are very finicky" means. CI tests passed despite that bug so we should add a test for `0. in (0. .* (1:10))`.

Reply:
`iszero(step(r)) && return true` changes behavior when `first(r) < x <= last(r)` or `last(r) <= x < first(r)` and `iszero(step(r))` to say that `x in r`. I think this is okay because 1) ranges have lengths and a range with `first(r) ≠ last(r)` and `iszero(setp(r))` has no length and 2) a range with step 0, if it did exist, would be continuous, so any number between `first(x)` and `last(x)` should be in it.

Reply:
I‘m not sure I get it. if `first(r) != last(r) && iszero(step(r))` i'm assuming that `r` must be empty and would be catched with L1391?